src/request.rs

```diff
@@ -887,6 +885,47 @@ impl<'a> Request<'a> {
     ///let response = Request::new(&uri).send(&mut writer).unwrap();
     ///```
     pub fn send<T: Write>(&self, writer: &mut T) -> Result<Response, error::Error> {
+        if self.inner.uri.scheme() == "https" {
+            let mut cnf = tls::Config::default();
+            let cnf = match self.root_cert_file_pem {
+                Some(p) => cnf.add_root_cert_file_pem(p)?,
+                None => &mut cnf,
+            };
+            self.send_with_config(writer, Some(cnf))
+        } else {
+            self.send_with_config(writer, None)
+        }
+    }
+
+    /// Sends a HTTPs request. The connection will only be established if the supplied certificate
+    /// matches the server's root certificate.
+    /// Supply the content of the server's root certificate as a String. There are no predefined trusted
+    /// root certificates in the tls config.
+    pub fn send_with_pem_certificate<T: Write>(
+        &self,
+        writer: &mut T,
+        certificate_content: Option<String>,
+    ) -> Result<Response, error::Error> {
+        if self.inner.uri.scheme() == "https" {
+            let mut cnf = tls::Config::empty_root_store();
+            let cnf = match certificate_content {
+                Some(c) => cnf.add_root_cert_content_pem_file(&c)?,
+                None => &mut cnf,
+            };
+            self.send_with_config(writer, Some(cnf))
+        } else {
+            println!("Send with certificate, but not a https request!");
+            Err(error::Error::Tls)
+        }
+    }
+
+    /// Sends a HTTP request.
+    /// For https requests a tls configuration must be provided
+    pub fn send_with_config<T: Write>(
```

Comment:
Does this function need to be public (`pub`) ? I would've expected it to be private..

Reply:
To call a send request with a list of server's root certificates different from those currently proposed (empty or an old mozilla root's certificates list), without having to change the http_req lib. With such a method, we would not have needed to fork the lib